automation/compare-gems

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env ruby
+require 'bundler'
+require 'json'
+require 'open3'
+
+EXCLUDED_GROUPS = [:test, :development]
+SCL_PREFIXES = JSON.load(File.read(File.join(__dir__, '..', 'scl_prefixes.json')))
+
+def package_name(gem_name)
+  # TODO: EL8 has no SCL
+  "tfm-rubygem-#{gem_name}"
+end
+
+def get_upstream_versions(filename)
+  Bundler.settings.set_local(:without,  EXCLUDED_GROUPS)
+  Bundler.settings.set_local(:with, [])
+
+  definition = Bundler::Dsl.evaluate(filename, nil, true)
+  definition.resolve_remotely!
+
+  Hash[definition.requested_specs.map do |spec|
```

Comment:
the return value of `get_upstream_version` is a `Hash` built from the `compact`ed result of the `definition.requested_specs.map { block }`? not that readable imho ;)

Reply:
I miss dict comprehension